Fix test_cuda_primary_ctx being skipped on CUDA (non-ROCm)

lw opened the pull request:
Summary:
In https://github.com/pytorch/pytorch/pull/71146 that line was changed:
- before, the test would run on CUDA and skip on ROCm
- after, the test would skip on CUDA, skip on old ROCm and run on new ROCm

I believe the intention however was that it would run on CUDA, skip on old ROCm, and run on new ROCm.

Found by sypneiwski.

Test Plan:
eyes

By looking at the logs of the CI once it runs.

Differential Revision: D38582143

lw commented on torch/testing/_internal/common_utils.py:
```diff
@@ -983,24 +992,29 @@ def wrapper(*args, **kwargs):
     return dec_fn
 
+# Skips a test on CUDA if ROCm is unavailable or its version is lower than requested.
+def skipIfRocmVersionLessThan(version=None):
```
I wonder if this should be fully removed? The only other usage is https://github.com/pytorch/pytorch/blob/03abcf231726b094bd378ead59d81c673fe11b7d/test/test_nn.py#L11396

Does anyone know if that test should or shouldn't run on "real" CUDA?

albanD replied:
I think it is ok to keep the original name and change the behavior.

malfet replied:
But the behavior is very unexpected, to say the least..
I hope `@skipIfRocm` does not skip tests on CUDA

lw committed:
Fix test_cuda_primary_ctx being skipped on CUDA (non-ROCm) (#83182)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/83182

In https://github.com/pytorch/pytorch/pull/71146 that line was changed:
- before, the test would run on CUDA and skip on ROCm
- after, the test would skip on CUDA, skip on old ROCm and run on new ROCm

I believe the intention however was that it would run on CUDA, skip on old ROCm, and run on new ROCm.

Found by sypneiwski.

Test Plan:
eyes

By looking at the logs of the CI once it runs.

Differential Revision: D38582143

fbshipit-source-id: b17613ccff570b58f53110f9c32210d2079efe0b

albanD reviewed (approved):
SGTM thanks!
Let's make sure CI is green before merging!

malfet commented:
@pytorchbot merge -g
